.github/workflows/test_shell.yml

```diff
@@ -0,0 +1,27 @@
+name: Test Shellscripts (Conditional)
+on:
+  pull_request:
+    paths:
+      - '**/*.bats'
+      - '**/*.sh'
```

Comment:
Not convinced conditional run is the right thing to do here as it only takes 30 sec including install step, but I'm hesitant to add another dependency on the main test since it's already fairly cluttered.

Reply:
I think it should be part of the `npm run lint` command so devs just run one command and everything is linted, but it would be a shame to add more to the dev environment setup requirements. There is an [npm package](https://www.npmjs.com/package/shellcheck) that might be useful to do the installation for us. Might be worth looking in to...

Reply:
That's a good suggestion, I'm also making use of shfmt to list all shell scripts which is needed for shellcheck to find and test those files. I can't use `*.sh` and `*.bats` since not all of them have those extensions, but something like 

```shell
for f in $(git ls-files); do
    head -n1 $f | egrep -q '#!/bin/(ba)?sh' && echo $f;
done
```
might work as an alternative
(also need to consider scripts with /usr/bin/env bash and `*.bats` extensions so not exactly this script)